libmscore/sym.cpp

```diff
@@ -6586,6 +6591,69 @@ void initScoreFonts()
       QFont::insertSubstitution("Finale Broadway Text",  "MuseJazz Text");
       QFont::insertSubstitution("ScoreFont",      "Leland Text"); // alias for current Musical Text Font
       ScoreFont::fallbackFont();   // load fallback font
+
+      QString userFontsPath = preferences.getString(PREF_APP_PATHS_MYSCOREFONTS);
+      scanUserFonts(userFontsPath);
+      preferences.addOnSetListener([](const QString& key, const QVariant& value) {
+            if (key == PREF_APP_PATHS_MYSCOREFONTS) {
+                  scanUserFonts(value.toString());
+                  }
+            });
+      }
+
+void ScoreFont::scanUserFonts(const QString& path)
+      {
+      QVector<ScoreFont> userfonts;
+
+      QDirIterator iterator(path, QDir::Dirs | QDir::NoDotAndDotDot | QDir::Readable);
+
+      while (iterator.hasNext()) {
+          QString fontDir = iterator.next();
+          QString fontDirPath = iterator.filePath() + "/";
+          QString fontDirName = iterator.fileName();
+
+          QString fontName;
+          QString fontFilename;
+          QDirIterator innerIterator(fontDirPath, { "*.otf", "*.ttf" }, QDir::Files);
+
+          while (innerIterator.hasNext()) {
+              QString potentialFontFile = innerIterator.next();
+              QFileInfo fileinfo(potentialFontFile);
+
+              if (fileinfo.completeBaseName().toLower() == fontDirName.toLower()) {
+                  fontName = fileinfo.completeBaseName();
+                  fontFilename = innerIterator.fileName();
+                  break;
+              }
+          }
+
+          bool hasMetadataFile = QFileInfo::exists(fontDirPath + "metadata.json");
+
+          if (hasMetadataFile && !fontFilename.isEmpty()) {
+              QByteArray name = fontName.toLocal8Bit();
+              QByteArray dp = fontDirPath.toLocal8Bit();
+              QByteArray fn = fontFilename.toLocal8Bit();
+              userfonts << Ms::ScoreFont(name.data(), name.data(), dp.data(), fn.data());
+          }
+      }
+
+      qDebug() << "Found" << userfonts.count() << "user score fonts.";
+
+      // TODO: Check for fonts that duplicate built-in fonts
```

Comment:
Allowing duplicates might even be good, if the external one would override the internal one. That would allow for example to load a newer version of Bravura, Leland, without needing to rebuild.